### PR TITLE
Fix a C++20 cmake check.

### PR DESCRIPTION
**Pull request by bangerth**
I had a typo in the last commit in #14915. This fixes it.

### CODE REVIEW

File: cmake/checks/check_01_cxx_features.cmake [hunk 1]
```diff
@@ -81,10 +81,13 @@ macro(_test_cxx20_support)
     #  error \"insufficient support for C++20\"
     #endif
 
-    #if !(defined __cpp_type_identity) || (__cpp_lib_type_identity < 201806)
```
Comment (bangerth) [inline cmake/checks/check_01_cxx_features.cmake:84]: It should have been `__cpp_lib_type_identity`, not `__cpp_type_identity`.

File: cmake/checks/check_01_cxx_features.cmake [hunk 2]
```diff
@@ -81,10 +81,13 @@ macro(_test_cxx20_support)
...
-    #  error \"insufficient support for C++20\"
+    #if !(defined __cpp_lib_type_identity)
+    #  error \"insufficient support for C++20: __cpp_type_identity not defined\"
```
Comment (masterleinad) [inline cmake/checks/check_01_cxx_features.cmake:85]:
```suggestion
    #  error \"insufficient support for C++20: __cpp_lib_type_identity not defined\"
```

Comment (bangerth) [reply cmake/checks/check_01_cxx_features.cmake]: Ah yes, so addressed.

**Review by masterleinad — APPROVED** [review]

**Comment by bangerth** [comment]
It looks like the OSX check is stuck, but running all CI checks really seems unnecessary for this patch.

**Pull request merged**